### PR TITLE
Add section in tutorial for IterableDataset

### PR DESCRIPTION
**Pull request by stevhliu**
Introduces an `IterableDataset` and how to access it in the tutorial section. It also adds a brief next step section at the end to provide a path for users who want more explanation and a path for users who want something more practical and learn how to preprocess these dataset types. It'll complement the awesome new doc introduced in:

- #5410

### CODE REVIEW

File: docs/source/access.mdx [hunk 1]
```diff
@@ -78,4 +84,34 @@ Slicing returns a slice - or subset - of the dataset, which is useful for viewin
...
+>>> iterable_dataset = load_dataset("food101", split="train", streaming=True)
+>>> for example in iterable_dataset:
+...     print(example)
+...     break
+{'image': <PIL.JpegImagePlugin.JpegImageFile image mode=RGB size=384x512 at 0x7F0681F5C520>, 'label': 6}
+```
+
+An [`IterableDataset`] progressively iterates over a dataset one example at a time, so you don't have to wait for the whole dataset to download before you can use it. As you can imagine, this is quite useful for large datasets you want to use immediately!
+
+However, this means an [`IterableDataset`]'s behavior is different from a regular [`Dataset`]. You don't get random access to examples in an [`IterableDataset`]. Instead, you can call `next(iter()` to return the next item from the [`IterableDataset`]:
```
Comment (albertvillanova) [inline docs/source/access.mdx:105]: Nit: strictly speaking, elements can be accessed using the iterable protocol; not necessarily only `next(iter())`, also can be iterated using for loops (like in the example above).
```suggestion
However, this means an [`IterableDataset`]'s behavior is different from a regular [`Dataset`]. You don't get random access to examples in an [`IterableDataset`]. Instead, you should iterate over its elements, for example by calling `next(iter())` to return the next item from the [`IterableDataset`]:
```

**Review by albertvillanova — APPROVED** [review]
Thanks, @stevhliu. Great to have this introduction to `IterableDataset`s.

File: docs/source/access.mdx (hunk 1 above)
Comment (lhoestq) [reply docs/source/access.mdx:105]: I'd also suggest to use a for loop instead if mentioning `next()` here.

You can also show how to get the first 5 examples though:
```python
print(list(iterable_dataset.take(5)))
```

**Review by lhoestq — COMMENTED** [review]
Nice thank you !

**Pull request merged**